sklearn/ensemble/_hist_gradient_boosting/tests/test_binning.py

```diff
@@ -312,3 +317,140 @@ def test_infinite_values():
 
     expected_binned_X = np.array([0, 1, 2, 3]).reshape(-1, 1)
     assert_array_equal(bin_mapper.transform(X), expected_binned_X)
+
+
+@pytest.mark.parametrize("categorical", [None, np.array([False])])
+def test_categorical_transform_error_with_no_categoricals(categorical):
+    # Properties and
+    X = np.arange(10, dtype=float).reshape(-1, 1)
+    bin_mapper = _BinMapper(n_bins=5, categorical=categorical).fit(X)
+
+    assert len(bin_mapper.bin_categories_) == 0
+
+    msg = ("categorical_only=True can only be set when there are "
+           "categorical features")
+    with pytest.raises(ValueError, match=msg):
+        bin_mapper.transform(X, categorical_only=True)
+
+
+@pytest.mark.parametrize("n_bins", [7, 15, 256])
+def test_categorical_n_bins_greater_than_equal_cardinality(n_bins):
```

Comment:
I think the equal case is covered in the other test?

Reply:
Sorry, this isn't resolved.

There are 6 categories and you allow for 15. The equal case is, as previously mentioned, handled in `test_categorical_n_bins_less_than_cardinality`